plugins/modules/ilo_redfish_info.py

```diff
@@ -123,31 +197,50 @@ def main():
             username=dict(),
             password=dict(no_log=True),
             auth_token=dict(no_log=True),
-            timeout=dict(type='int', default=10)
+            timeout=dict(type='int'),
+            cert_file=dict(type='str'),
+            key_file=dict(type='str')
         ),
         required_together=[
             ('username', 'password'),
+            ('cert_file', 'key_file')
         ],
         required_one_of=[
-            ('username', 'auth_token'),
+            ('username', 'auth_token', 'cert_file'),
         ],
         mutually_exclusive=[
-            ('username', 'auth_token'),
+            ('username', 'auth_token', 'cert_file'),
         ],
         supports_check_mode=True
     )
 
+    if not HAS_iLO_REDFISH:
+        module.fail_json(msg="missing required fucntions in ilo_redfish_utils.py")
+
     creds = {"user": module.params['username'],
              "pswd": module.params['password'],
              "token": module.params['auth_token']}
 
     timeout = module.params['timeout']
+    if module.params['timeout'] is None:
+        timeout = 10
+        module.deprecate(
+            'The default value {0} for parameter param1 is being deprecated and it will be replaced by {1}'.format(
+                10, 60
+            ),
+            version='8.0.0',
+            collection_name='community.general'
+        )
 
     root_uri = "https://" + module.params['baseuri']
+
+    if module.params["cert_file"]:
+        creds["token"] = ilo_certificate_login(root_uri, module, module.params["cert_file"], module.params["key_file"])
+
     rf_utils = iLORedfishUtils(creds, root_uri, timeout, module)
 
     # Build Category list
-    if "all" in module.params['category']:
+    if module.params['category'] == "all":
```

Comment:
The parameter `category` is not provided with a fixed list of choices, and it is of type `list`. I don't think this is right.

Reply:
Hi, existing redfish_info module seems to be following along the same lines.
`if "all" in module.params['category']:
        for entry in CATEGORY_COMMANDS_ALL:
            category_list.append(entry)
    else:
        # one or more categories specified
        category_list = module.params['category']`

https://github.com/ansible-collections/community.general/blob/main/plugins/modules/redfish_info.py

We followed what was already present as an example. Should we be changing it? Please advice.

Reply:
Any update on this?

Reply:
Your change is deviating from what's in the common Redfish module. Your change is treating it like a string rather than a list.